src/Plugins/AntiW3Schools.php

```diff
@@ -0,0 +1,130 @@
+        'jquery' => '[Check the official jQuery API Documentation](http://api.jquery.com/) for help with the jQuery library.',
+        'angular' => '[Check the official AngularJS API Documentation](https://docs.angularjs.org/api) for help with the AngularJS framework.',
+        'xml'=> '[Check W3\'s documentation on XML](https://www.w3.org/standards/xml/core) for more information.',
+        'ajax' => '[Check the Mozilla Developer Network AJAX documentation](https://developer.mozilla.org/docs/AJAX) for more information.',
+    ];
+
+    private $chatClient;
+
+    public function __construct(ChatClient $chatClient)
+    {
+        $this->chatClient = $chatClient;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getName(): string
+    {
+        return 'AntiW3Schools';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getDescription(): string
+    {
+        return 'A small monitoring plugin which bothers people when a w3schools link is detected.';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getMessageHandler(): array
+    {
+        return [$this, 'handleMessage'];
+    }
+
+    /**
+     * Entry point for checking a message for w3schools links.
+     * @param Message $message
+     * @return Promise
+     */
+    public function handleMessage(Message $message): Promise
+    {
+        return $this->containsTerribleUri($message->getText())
+            ? $this->chatClient->postReply($message, $this->createReply($message))
+            : new Success();
+    }
+
+    /**
+     * Check if text contains a W3C link.
+     * @param string $text
+     * @return bool
+     */
+    private function containsTerribleUri(string $text) : bool
+    {
+        return preg_match('/' . self::BAD_HOST_PATTERN . '/i', $text) === 1;
+    }
+
+    /**
+     * Create a suitable message to bother people about W3C.
+     * @param Message $message
+     * @return string
+     */
+    private function createReply(Message $message) : string
+    {
+        $return = 'W3Schools should not be trusted as a reliable resource. [Click here to read why](http://www.w3fools.com/).';
+
+        $categories = $this->dissectCategoriesFromString($message->getText());
+
+        foreach($categories as $category) {
+            $return .= ' ' . self::W3S_CATEGORY_RESPONSES[$category];
+        }
+
+        return $return;
+
+    }
+
+    /**
+     * Retrieve W3S categories from a given string.
+     * @param string $text
+     * @return string[]
+     */
+    private function dissectCategoriesFromString(string $text) {
+
+        // Extract the category from all W3S URIs found.
+        $categoryNames = array_keys(self::W3S_CATEGORY_RESPONSES);
+
+        $matchSets = [];
+
+        if(
+            !preg_match_all(
+                '/' . self::BAD_HOST_PATTERN . '\/(' . implode('|', $categoryNames) . ')\//i',
```

Comment:
Since you need to match literal slashes, why not use a different delimiter? 

e.g. `'#' . self::BAD_HOST_PATTERN . '/(' . implode('|', $categoryNames) . ')/#i'`